src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdaterTest.java

```diff
@@ -330,4 +130,74 @@ public void should_not_update_case_data_without_any_documents() {
+        LocalDateTime exScannedDate,
+        LocalDateTime exDeliveryDate
+    ) {
+        assertThat(resultScannedDocuments.get(0).value.type).isEqualTo("type_A");
+        assertThat(resultScannedDocuments.get(0).value.subtype).isEqualTo("subtype_A");
+        assertThat(resultScannedDocuments.get(0).value.scannedDate).isEqualTo(caseScannedDate);
+        assertThat(resultScannedDocuments.get(0).value.deliveryDate).isEqualTo(caseDeliveryDate);
+        assertThat(resultScannedDocuments.get(0).value.fileName).isEqualTo("file_name_AA");
+        assertThat(resultScannedDocuments.get(0).value.exceptionRecordReference).isEqualTo("exceptionRecordReference");
+        assertThat(resultScannedDocuments.get(0).value.controlNumber).isEqualTo("control_number_A");
+        assertThat(resultScannedDocuments.get(0).value.document.url).isEqualTo("file://file_A");
+        assertThat(resultScannedDocuments.get(0).value.document.filename).isEqualTo("file_name_A");
+        assertThat(resultScannedDocuments.get(0).value.document.binaryUrl).isEqualTo("binary_url_A");
+
+        assertThat(resultScannedDocuments.get(1).value.type).isEqualTo("Form_1");
+        assertThat(resultScannedDocuments.get(1).value.subtype).isEqualTo("subtype_1");
+        assertThat(resultScannedDocuments.get(1).value.scannedDate).isEqualTo(exScannedDate);
+        assertThat(resultScannedDocuments.get(1).value.deliveryDate).isEqualTo(exDeliveryDate);
+        assertThat(resultScannedDocuments.get(1).value.fileName).isEqualTo("file_name_11");
+        assertThat(resultScannedDocuments.get(1).value.exceptionRecordReference).isNull();
+        assertThat(resultScannedDocuments.get(1).value.controlNumber).isEqualTo("control_number_1");
+        assertThat(resultScannedDocuments.get(1).value.document.url).isEqualTo("file://file_1");
+        assertThat(resultScannedDocuments.get(1).value.document.filename).isEqualTo("file_name_1");
+        assertThat(resultScannedDocuments.get(1).value.document.binaryUrl).isEqualTo("binary_url_1");
+
+
+        assertThat(resultScannedDocuments.get(2).value.type).isEqualTo("Form_2");
+        assertThat(resultScannedDocuments.get(2).value.subtype).isEqualTo("subtype_2");
+        assertThat(resultScannedDocuments.get(2).value.scannedDate).isEqualTo(exScannedDate);
+        assertThat(resultScannedDocuments.get(2).value.deliveryDate).isEqualTo(exDeliveryDate);
+        assertThat(resultScannedDocuments.get(2).value.fileName).isEqualTo("file_name_22");
+        assertThat(resultScannedDocuments.get(2).value.exceptionRecordReference).isNull();
+        assertThat(resultScannedDocuments.get(2).value.controlNumber).isEqualTo("control_number_2");
+        assertThat(resultScannedDocuments.get(2).value.document.url).isEqualTo("file://file_2");
+        assertThat(resultScannedDocuments.get(2).value.document.filename).isEqualTo("file_name_2");
+        assertThat(resultScannedDocuments.get(2).value.document.binaryUrl).isEqualTo("binary_url_2");
```

Comment:
perhaps instead of using some magic strings those assertions could use values from an object passed to this method?

Reply:
That was my thought too - same issue as with InputHelper. It needs to be clear why we expect those values. It won't be clear if they're defined in multiple places. Just like Rafał said, best to pass them in some objects, as making them static won't help.